test/opt/test_types.cpp

```diff
@@ -218,4 +218,20 @@ TEST(Types, AllTypes) {
   }
 }
 
+#define TestWidthOfType(ty, ...)                                      \
+  TEST(Types, TypeWidthOf##ty) {                                      \
+    std::vector<uint32_t> widths = {1, 2, 4, 8, 16, 32, 48, 64, 128}; \
+    std::vector<std::unique_ptr<ty>> types;                           \
+    for (uint32_t w : widths) {                                       \
+      types.emplace_back(new ty(w, ##__VA_ARGS__));                   \
```

Comment:
Ha, I'm afraid this `##__VA_ARGS__` trick suffers on some old VS compilers? (Although we only test >= VS2013 in buildbot, it's still better to be able to compile on older compilers.)

I'm not opposed to duplicate the code for Integer & Float anyway, it's just two cases.